src/ios/BEMShakeNotifier.m

```diff
@@ -0,0 +1,431 @@
+        return;
+    }
+    
+    __block NSDictionary* localNotifyConfig = command.arguments[1];
+    
+    if (isNSNull(localNotifyConfig) || [localNotifyConfig count] == 0) {
+        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"localNotifyConfig is null or empty"];
+        [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+        return;
+    }
+    
+    // enabling the listener means removing an entry from the muted list
+    [self removeEntry:localNotifyConfig
+             forEvent:eventName
+            withLabel:MUTED_LIST_KEY];
+    pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
+    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+}
+
+- (void)disableEventListener:(CDVInvokedUrlCommand*)command
+{
+    CDVPluginResult* pluginResult;
+    
+    __block NSString* eventName = command.arguments[0];
+    
+    if (isNSNull(eventName) || eventName == nil || [eventName length] == 0) {
+        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"eventName is null"];
+        [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+        return;
+    }
+    
+    __block NSDictionary* localNotifyConfig = command.arguments[1];
+    
+    if (isNSNull(localNotifyConfig) || localNotifyConfig == nil || [localNotifyConfig count] == 0) {
+        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"localNotifyConfig is null or empty"];
+        [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+        return;
+    }
+    
+    // disabling the listener means adding an entry to the muted list
+    [self addOrReplaceEntry:localNotifyConfig
+                   forEvent:eventName
+                  withLabel:MUTED_LIST_KEY];
+    pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
+    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+}
+
+-(void)enableSensor{
+    motionManager = [[CMMotionManager alloc] init];
+    if([motionManager isAccelerometerAvailable]){
+        if([motionManager isAccelerometerActive] == NO){
+            [LocalNotificationManager addNotification:[NSString stringWithFormat:
+                                                       @"DetectedShake"]
+                                                                    showUI:FALSE];
+            motionManager.accelerometerUpdateInterval = 0.1f;
+            __block double currentAcceleration = 0.0;
+            __block double acceleration = 0.0;
+            __block double lastUpdated = 0.0;
+            
+            double yAccelerationConst = 0.4;
+            [motionManager startAccelerometerUpdatesToQueue:[NSOperationQueue mainQueue] withHandler:^(CMAccelerometerData *data, NSError *error){
```

Comment:
Does this actually collect data while in the background?

Reply:
apparently it does so if your app is already launched for other background operations.
https://stackoverflow.com/questions/21947864/getting-accelerometer-data-in-background-using-coremotion
https://stackoverflow.com/questions/8716466/how-nike-gps-on-iphone-receives-accelerometer-updates-in-the-background
For some time.
https://forums.developer.apple.com/thread/65074